hibernate-core/src/test/java/org/hibernate/orm/test/mapping/lazytoone/polymorphic/JoinFetchedPolymorphicToOneTests.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.mapping.lazytoone.polymorphic;
+
+import java.math.BigDecimal;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import org.hibernate.Hibernate;
+import org.hibernate.annotations.Fetch;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.AvailableSettings;
+
+import org.hibernate.testing.bytecode.enhancement.BytecodeEnhancerRunner;
+import org.hibernate.testing.bytecode.enhancement.EnhancementOptions;
+import org.hibernate.testing.jdbc.SQLStatementInterceptor;
+import org.hibernate.testing.junit4.BaseNonConfigCoreFunctionalTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static javax.persistence.FetchType.LAZY;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hibernate.annotations.FetchMode.JOIN;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Steve Ebersole
+ */
+@RunWith( BytecodeEnhancerRunner.class)
+@EnhancementOptions( lazyLoading = true )
+public class JoinFetchedPolymorphicToOneTests extends BaseNonConfigCoreFunctionalTestCase {
+	@Test
+	public void testInheritedToOneLaziness() {
+		inTransaction(
+				(session) -> {
+					sqlStatementInterceptor.clear();
+
+					final Order order = session.byId( Order.class ).getReference( 1 );
+					assertThat( sqlStatementInterceptor.getQueryCount(), is( 0 ) );
+
+					System.out.println( "Order # " + order.getId() );
+					assertThat( sqlStatementInterceptor.getQueryCount(), is( 0 ) );
+
+					System.out.println( "  - amount : " + order.getAmount() );
+					// triggers load of base fetch state
+					assertThat( sqlStatementInterceptor.getQueryCount(), is( 1 ) );
+
+					final Customer customer = order.getCustomer();
+					// customer is part of base fetch state
```

Comment:
Again, a confusing combination of `@ManyToOne( fetch = LAZY )` and `@Fetch( JOIN )`. Shouldn't the access to the customer trigger the loading of the customer?

Reply:
Same conversation as above